pkg/raft/rawnode.go

```diff
@@ -63,19 +63,32 @@ func NewRawNode(config *Config) (*RawNode, error) {
 	return rn, nil
 }
 
-// Tick advances the internal logical clock by a single tick.
+// Tick advances the internal logical clock by a single tick. Election timeouts
+// and heartbeat timeouts are in units of ticks.
 func (rn *RawNode) Tick() {
 	rn.raft.tick()
 }
 
-// Campaign causes this RawNode to transition to candidate state.
+// Campaign causes this RawNode to transition to candidate state and start
+// campaigning to become leader.
 func (rn *RawNode) Campaign() error {
 	return rn.raft.Step(pb.Message{
 		Type: pb.MsgHup,
 	})
 }
 
-// Propose proposes data be appended to the raft log.
+// Propose proposes an entry with the given data to be appended to the raft log.
+//
+// Returns ErrProposalDropped if the proposal could not be made. A few reasons
+// why this can happen:
+//
+//   - the RawNode is not the leader, and DisableProposalForwarding is true
+//   - the RawNode is transferring the leadership away
+//   - the proposal overflows the internal size limits
+//   - the proposal is incorrect for other reasons
+//
+// If the proposal is submitted, it can still be lost or not committed. It is
+// the user's job to retry the proposal in this case.
```

Comment:
it could be worth mentioning that a proposal can also be dropped silently (leadership changes, etc), so one needs to be prepared to retry proposals even if no error is returned here.

Reply:
Done.